tests/unit-tests/importer/tests-donations.php

```diff
@@ -0,0 +1,407 @@
+
+		$this->total = $this->importer_class->get_csv_data_from_file_dir( $this->csv_file );
+
+		parent::setUp();
+	}
+
+	/**
+	 * Tear it down.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+	}
+
+	/**
+	 * Get CSV mapped items.
+	 *
+	 * @since 2.1
+	 * @return array
+	 */
+	private function get_csv_mapped_items() {
+		return array(
+			'form_title',
+			'amount',
+			'currency',
+			'form_level',
+			'post_date',
+			'first_name',
+			'last_name',
+			'company_name',
+			'email',
+			'mode',
+			'post_status',
+			'gateway',
+			'notes',
+			'line1',
+			'line2',
+			'city',
+			'zip',
+			'state',
+			'country',
+			'post_meta',
+		);
+	}
+
+	/**
+	 * Get import setting
+	 *
+	 * @return array
+	 */
+	private function get_import_setting() {
+
+		return $import_setting = array(
+			'delimiter'   => 'csv',
+			'mode'        => 0,
+			'create_user' => 1,
+			'delete_csv'  => 0,
+			'per_page'    => 25,
+			'raw_key'     => $this->get_csv_mapped_items(),
+		);
+	}
+
+	public function import_donation_in_dry_run() {
+		$import_setting = $this->import_setting;
+
+		$raw_key = $import_setting['raw_key'];
+
+		// data from CSV
+		$raw_data = $this->raw_data;
+
+		// donation meta key name
+		$main_key = $this->main_key;
+
+		// first add donation in dry run mode
+		$import_setting['dry_run'] = 1;
+
+		if ( ! empty( $import_setting['dry_run'] ) ) {
+			$import_setting['csv_raw_data'] = $raw_data;
+
+			$import_setting['donors_list'] = Give()->donors->get_donors( array(
+				'number' => - 1,
+				'fields' => array( 'id', 'user_id', 'email' ),
+			) );
+		}
+
+		$current_key = 1;
+		foreach ( $raw_data as $row_data ) {
+			$import_setting['donation_key'] = $current_key;
+			$payment_id                     = give_save_import_donation_to_db( $raw_key, $row_data, $main_key, $import_setting );
+			$current_key ++;
+		}
+	}
+
+	/**
+	 * Test by uploading live donation from CSV
+	 *
+	 * @since 2.1
+	 */
+	public function import_donation_in_live( $import_setting = array() ) {
+		give_import_donation_report_reset();
+
+		if ( empty( $import_setting ) ) {
+			$import_setting = $this->get_import_setting();
+		}
+
+		$raw_key = $import_setting['raw_key'];
+
+		$file_dir = $this->csv_file;
+
+		// get the total number of rom from CSV
+		$total = $this->importer_class->get_csv_data_from_file_dir( $file_dir );
+
+		// get data from CSV
+		$raw_data = give_get_raw_data_from_file( $file_dir, 1, $total, ',' );
+		$main_key = give_get_raw_data_from_file( $file_dir, 0, 1, ',' );
+		$main_key = $main_key[0];
+
+		$current_key = 1;
+		foreach ( $raw_data as $row_data ) {
+			$import_setting['donation_key'] = $current_key;
+			$payment_id                     = give_save_import_donation_to_db( $raw_key, $row_data, $main_key, $import_setting );
+			$current_key ++;
+		}
+	}
+
+	/**
+	 * Get the total number of row from the CSV and count
+	 * there are total 11 row 10 donation and 1st one is Donation key name
+	 */
```

Comment:
Some methods have @since and some don't. Be consistent.

Reply:
Added since in all the functions of donation importer test case class